### PR TITLE
Setup script

### PR DESCRIPTION
**Pull request by louisj381**
Base Setup Script for a fresh image of Raspbian Linux, to configure drivers, install rabbit.mq, download QT naively to the PI and setup the toolchain.

### CODE REVIEW

File: SetupScript.sh
Comment (ExtraGravity) [inline SetupScript.sh]: Do this in opt

File: SetupScript.sh
Comment (ExtraGravity) [inline SetupScript.sh]: Concatenate the `cd` commands

File: SetupScript.sh
Comment (ExtraGravity) [inline SetupScript.sh]: `sudo` hurts the portability and ability to automate these scripts. It's better to require sudo to run the script.

File: SetupScript.sh
Comment (ExtraGravity) [inline SetupScript.sh]: @bill-luu this was in another step wasn't it?

**Review by ExtraGravity — CHANGES_REQUESTED** [review]

File: SetupScript.sh
Comment (bill-luu) [reply SetupScript.sh]: Yes! Lets keep the installation of Epsilon-Dashboard & other programs in a separate install script.

For rabbitmq setup, you can just copy/paste the necessary stuff from EpsilonDashboard.sh

File: SetupScript.sh
Comment (bill-luu) [inline SetupScript.sh]: Does having 3 #s do anything? Otherwise I think you only need one.

File: SetupScript.sh
Comment (bill-luu) [inline SetupScript.sh]: I would prefer using the flags `-y --force-yes` instead of piping it.

File: SetupScript.sh
Comment (bill-luu) [inline SetupScript.sh]: Address Enoch's comment that this should be in `/opt/`

**Review by bill-luu — CHANGES_REQUESTED** [review]
What's the scope of this PR? Do you want this merged before or after the QT grind?

File: SetupScript.sh
Comment (bill-luu) [inline SetupScript.sh]: -y is actually a flag of apt-get, so thebcommand would be apt-get --yes --force-yes

--force-yes is there to get rid of some other prompt about keyrings.

**Review by bill-luu — CHANGES_REQUESTED** [review]

**Comment by ExtraGravity** [comment]
@bill-luu  I'll review after your approval. I would like to have final approval on this though since it's pretty important.

**Comment by louisj381** [comment]
Note: Changed local hostname to "pi@primarypi", patch file to be added to next commit regarding setup-script

**Comment by louisj381** [comment]
Note that the patch works if within the QT5 directory, there is the version of init-repository that comes from v5.5.1(git checkout v5.5.1), and the patch file is moved to the ~/opt directory
https://github.com/UCSolarCarTeam/Epsilon-Raspberry/pull/5/commits/b43bbc9b3b3fffde4f8c76e5dd62e442c023e7b6

**Comment by louisj381** [comment]
Currently, the ./configure command prompts confirmation of QT licensing, where you have to respond "yes" to question for configuration to begin.
https://github.com/UCSolarCarTeam/Epsilon-Raspberry/pull/5/commits/024c09c4efbf219f62931e2baa0793706856fcf5#diff-c10aed41ae608b776005a559dd5fd739

**Comment by louisj381** [comment]
Below are screenshots of the build options after configuration:
build_options_1
build_options_2
build_options_3

**Comment by bill-luu** [comment]
At first glance the configure output looks good..... might even fix the Music Player issue we were having!

**Comment by louisj381** [comment]
I'm guessing the problem with the Epsilon-Dashboard.sh script is the test expression to check if RabbitMQ is already installed or not. When I ran the script, it said that rabbitMQ was already installed however I did not have the SimpleAmqpClient repo cloned in my /tmp directory. Therefore most of the dependencies were not installed. Once that was complete and my LD_LIBRARY_PATH was set appropriately, the executable ran on the pi. I am sshing to the pi, so I'm guessing that is why this message popped up when running....and looped through the terminal forever until I killed the process:
connection_issue

**Comment by bill-luu** [comment]
That's expected output for the dashboard at the moment, it's trying to connect to a rabbitmq exchange named `""`, and can't find one. That'll be something to fix in the dashboard!

**Comment by bill-luu** [comment]
README has merge conflicts

**Comment by louisj381** [comment]
Should be ready for final pass-through @enochtsang & @bill-luu

**Comment by ExtraGravity** [comment]
Can you make this a branch off the main repo? Forks are annoying to work with.

**Pull request closed without merging**
